service/namespace.go

```diff
@@ -14,7 +15,7 @@ const resultChannel string = "Result"
 
 // namespace returns the namespace of the service.
 func (service *Service) namespace() []string {
-	h, err := hex.DecodeString(service.Hash)
+	h, err := base58.Decode(service.Hash)
 	if err != nil {
 		panic(err)
```

Comment:
can't we simplify this method and solve this panic at the same time?

If I remember we were decoding the and re-encoding in sha1 just because of the length that was too long for docker but here base58 is short enough so this could be removed.

Also we have `service.Hash` which is already encoded in base58 so this function could simply be `return service.Hash`, maybe even `service.Hash[:xxx]` with xxx the maximum length allowed by docker and in this case we simplify this function and we don't need to delete the previous services (just they might have conflicts if you are really unlucky)

Reply:
we can but as I remember it required keeping hash as bytes first (and it's not the case in this PR)

Reply:
not sure how it is related to the bytes here. I'm just saying that here we are decoding and re-encoding, we could send the encoded value directly (with a limit on the size).
```go
func (service *Service) namespace() []string {
  return service.Hash[:xxx]
}
```

With the hash as byte it will require encoding here but for this PR we already have the encoded version in database

Reply:
you can't take part in the hash as it's not good practice (and it doesn't guarantee uniqueness)  - this is why sha1 sum is calculated here and not taking a substring of hash.

Reply:
Done

Reply:
I still think that we could go a step further with like I say just returning the Hash especially that this is what we are doing on the latest release
https://github.com/mesg-foundation/core/blob/348aa982644ff31e9571813b298919dd99ed1c85/service/namespace.go#L13-L15
So this will not break anything and we don't have to cap anything because that was never released

Reply:
We can't because of service.Hash will be too long for docker label (this is why we using sha1 here)

Reply:
this is the namespace of the service anyway not docker. Also right now the hash for docker is something like

`core_hashxxxxxxx_service`

And I think sha1 is 40 characters and base58 is 44 so this is short enough and anyway this sha for docker should be done in docker but that's a separated PR.